op-devstack/stack/context.go

```diff
@@ -104,22 +42,13 @@ func ContextWithID(ctx context.Context, id slog.LogValuer) context.Context {
 	if idWithKind, ok := id.(KindProvider); ok {
 		ctx = ContextWithKind(ctx, idWithKind.Kind())
 	}
-	ctx = context.WithValue(ctx, idCtxKey, id)
-	ctx = log.RegisterLogAttrOnContext(ctx, "id", idCtxKey)
+	ctx = log.AddLogAttrToContext(ctx, "id", id)
 	return ctx
 }
 
-// IDFromContext retrieves a typed component ID from the context,
-// that was previously attached with ContextWithID.
-// If the ID is not present, or not of the same type, then a zero value is returned.
-func IDFromContext[I slog.LogValuer](ctx context.Context) (out I) {
```

Comment:
I think we should keep the `XFromContext` functions. See comments above about type safety. I think the log key should not be up to the caller. And I think we should not overly rely on the log-context registering to make these context functions do their work.

These context attributes can be useful in other scenarios than logging.

E.g. when tracing, or just adding a debugging case/breakpoint where you need to log if it's a certain service or thing.

Reply:
We should add them back when they're necessary, or I can add a generic function to do the same thing. Not sure what the advantage is of not using a generic function in this case.